### DISCUSSION TITLE
Does Nitro polyfill Node APIs like fetch?

### DISCUSSION THREAD

**Discussion by sannajammeh** [nitrojs/nitro#1883] [Q&A] [answered]
Hi! We are building apps with Nitro and run all systems on Node v20 and v21. These have builtin support for fetch via Undici. Will Nitro polyfill this with node-fetch or similar? There is very little information about the internals of Nitro on the docs page.

I am asking as the native Node.js fetch has support for Undici's CacheStorage API allowing us to cache things like JWKs from authorization servers at the runtime level.

**Comment by Hebilicious** [comment, marked as answer]
Nitro uses https://github.com/unjs/ofetch, which will use `globalThis.fetch` when available.
